overlord/snapstate/handlers.go

```diff
@@ -1369,6 +1370,7 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) (err erro
 			// This task is only used for unlinking a snap during refreshes so we
 			// can safely hard-code this condition here.
 			RunInhibitHint: runinhibit.HintInhibitedForRefresh,
+			SkipBinaries:   refreshAppAwarenessEnabled,
```

Comment:
unlink-current-snap is also used by MigrateHome, we need to make sure that  that is using the inhibit lock, I don't remember if it does or otherwise we probably do want to remove binaries in that usage, so we should check for something to make the distinction